// @clr/angular/esm2015/utils/focus/focus.service.js
/*
 * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import * as tslib_1 from "tslib";
import { Injectable, Optional, Renderer2, SkipSelf } from '@angular/core';
import { isObservable, of } from 'rxjs';
import { ArrowKeyDirection } from './arrow-key-direction.enum';
let FocusService = class FocusService {
    constructor(renderer) {
        this.renderer = renderer;
    }
    get current() {
        return this._current;
    }
    reset(first) {
        this._current = first;
    }
    listenToArrowKeys(el) {
        // The following listeners return false when there was an action to take for the key pressed,
        // in order to prevent the default behavior of that key.
        this.renderer.listen(el, 'keydown.arrowup', () => !this.move(ArrowKeyDirection.UP));
        this.renderer.listen(el, 'keydown.arrowdown', () => !this.move(ArrowKeyDirection.DOWN));
        this.renderer.listen(el, 'keydown.arrowleft', () => !this.move(ArrowKeyDirection.LEFT));
        this.renderer.listen(el, 'keydown.arrowright', () => !this.move(ArrowKeyDirection.RIGHT));
    }
    registerContainer(el) {
        this.container = el;
        this.renderer.setAttribute(el, 'tabindex', '0');
        this.listenToArrowKeys(el);
        // The following listeners return false when there was an action to take for the key pressed,
        // in order to prevent the default behavior of that key.
        this.renderer.listen(el, 'keydown.space', () => !this.activateCurrent());
        this.renderer.listen(el, 'keydown.enter', () => !this.activateCurrent());
    }
    moveTo(item) {
        this.renderer.setAttribute(this.container, 'aria-activedescendant', item.id);
        if (this.current) {
            this.current.blur();
        }
        item.focus();
        this._current = item;
    }
    /**
     * The second parameter, optional, is here to allow recursion to skip disabled items.
     */
    move(direction, current = this.current) {
        if (current) {
            const next = current[direction];
            if (next) {
                // Turning the value into an Observable isn't great, but it's the fastest way to avoid code duplication.
                // If performance ever matters for this, we can refactor using additional private methods.
                const nextObs = isObservable(next) ? next : of(next);
                nextObs.subscribe(item => {
                    if (item.disabled) {
                        return this.move(direction, item);
                    }
                    else {
                        this.moveTo(item);
                        return true;
                    }
                });
            }
        }
        return false;
    }
    activateCurrent() {
        if (this.current && this.current.activate) {
            this.current.activate();
            return true;
        }
        return false;
    }
};
FocusService = tslib_1.__decorate([
    Injectable(),
    tslib_1.__metadata("design:paramtypes", [Renderer2])
], FocusService);
export { FocusService };
export function clrFocusServiceFactory(existing, renderer) {
    return existing || new FocusService(renderer);
}
export const FOCUS_SERVICE_PROVIDER = {
    provide: FocusService,
    useFactory: clrFocusServiceFactory,
    deps: [[new Optional(), new SkipSelf(), FocusService], Renderer2],
};
