// @clr/angular/esm5/utils/focus/focus.service.js
/*
 * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import * as tslib_1 from "tslib";
import { Injectable, Optional, Renderer2, SkipSelf } from '@angular/core';
import { isObservable, of } from 'rxjs';
import { ArrowKeyDirection } from './arrow-key-direction.enum';
var FocusService = /** @class */ (function () {
    function FocusService(renderer) {
        this.renderer = renderer;
    }
    Object.defineProperty(FocusService.prototype, "current", {
        get: function () {
            return this._current;
        },
        enumerable: true,
        configurable: true
    });
    FocusService.prototype.reset = function (first) {
        this._current = first;
    };
    FocusService.prototype.listenToArrowKeys = function (el) {
        var _this = this;
        // The following listeners return false when there was an action to take for the key pressed,
        // in order to prevent the default behavior of that key.
        this.renderer.listen(el, 'keydown.arrowup', function () { return !_this.move(ArrowKeyDirection.UP); });
        this.renderer.listen(el, 'keydown.arrowdown', function () { return !_this.move(ArrowKeyDirection.DOWN); });
        this.renderer.listen(el, 'keydown.arrowleft', function () { return !_this.move(ArrowKeyDirection.LEFT); });
        this.renderer.listen(el, 'keydown.arrowright', function () { return !_this.move(ArrowKeyDirection.RIGHT); });
    };
    FocusService.prototype.registerContainer = function (el) {
        var _this = this;
        this.container = el;
        this.renderer.setAttribute(el, 'tabindex', '0');
        this.listenToArrowKeys(el);
        // The following listeners return false when there was an action to take for the key pressed,
        // in order to prevent the default behavior of that key.
        this.renderer.listen(el, 'keydown.space', function () { return !_this.activateCurrent(); });
        this.renderer.listen(el, 'keydown.enter', function () { return !_this.activateCurrent(); });
    };
    FocusService.prototype.moveTo = function (item) {
        this.renderer.setAttribute(this.container, 'aria-activedescendant', item.id);
        if (this.current) {
            this.current.blur();
        }
        item.focus();
        this._current = item;
    };
    /**
     * The second parameter, optional, is here to allow recursion to skip disabled items.
     */
    FocusService.prototype.move = function (direction, current) {
        var _this = this;
        if (current === void 0) { current = this.current; }
        if (current) {
            var next = current[direction];
            if (next) {
                // Turning the value into an Observable isn't great, but it's the fastest way to avoid code duplication.
                // If performance ever matters for this, we can refactor using additional private methods.
                var nextObs = isObservable(next) ? next : of(next);
                nextObs.subscribe(function (item) {
                    if (item.disabled) {
                        return _this.move(direction, item);
                    }
                    else {
                        _this.moveTo(item);
                        return true;
                    }
                });
            }
        }
        return false;
    };
    FocusService.prototype.activateCurrent = function () {
        if (this.current && this.current.activate) {
            this.current.activate();
            return true;
        }
        return false;
    };
    FocusService = tslib_1.__decorate([
        Injectable(),
        tslib_1.__metadata("design:paramtypes", [Renderer2])
    ], FocusService);
    return FocusService;
}());
export { FocusService };
export function clrFocusServiceFactory(existing, renderer) {
    return existing || new FocusService(renderer);
}
export var FOCUS_SERVICE_PROVIDER = {
    provide: FocusService,
    useFactory: clrFocusServiceFactory,
    deps: [[new Optional(), new SkipSelf(), FocusService], Renderer2],
};
